solr/core/src/test/org/apache/solr/search/TestSolrQueryParser.java

```diff
@@ -1839,24 +1842,55 @@ public void testFieldExistsQueries() throws SyntaxError {
                     + query
                     + "\".",
                 createdQuery instanceof ConstantScoreQuery);
-            assertTrue(
-                "PointField with NaN values must include \"exists or NaN\" if the field doesn't have norms or docValues: \""
-                    + query
-                    + "\".",
-                ((ConstantScoreQuery) createdQuery).getQuery() instanceof BooleanQuery);
-            assertEquals(
-                "PointField with NaN values must include \"exists or NaN\" if the field doesn't have norms or docValues: \""
-                    + query
-                    + "\". This boolean query must be an OR.",
-                1,
-                ((BooleanQuery) ((ConstantScoreQuery) createdQuery).getQuery())
-                    .getMinimumNumberShouldMatch());
-            assertEquals(
-                "PointField with NaN values must include \"exists or NaN\" if the field doesn't have norms or docValues: \""
-                    + query
-                    + "\". This boolean query must have 2 clauses.",
-                2,
-                ((BooleanQuery) ((ConstantScoreQuery) createdQuery).getQuery()).clauses().size());
+            if (schemaField.getType().isPointField()) {
```

Comment:
I really dislike these tests!  Sorry.  Instead of testing the implementation, it should test the functional effect.  Needn't test every combo of course but some special values with a bit of randomization.